app/controllers/foreman_tasks/api/tasks_controller.rb

```diff
@@ -234,9 +236,31 @@ def pagination_scope(scope, search_params)
         scope.limit(per_page).offset((page.to_i - 1) * per_page.to_i)
       end
 
-      def ordering_scope(scope, ordering_params)
+
+def ordering_scope(scope, ordering_params)
+  sort_by = ordering_params[:sort_by] || 'started_at'
+  sort_order = ordering_params[:sort_order] || 'DESC'
+  scope.select("*, COALESCE(#{Task.table_name}.ended_at, CURRENT_TIMESTAMP) - #{Task.table_name}.started_at as duration")
+  scope = scope.order("#{sort_by} #{sort_order}")
+end
+      def ordering_scope3(scope, ordering_params)
```

Comment:
Maybe we could figure out better names for `ordering_scope`, `ordering_scope1` and `ordering_scope3`?

Reply:
My PRs got mixed up and this was a draft for a different one, will clean this PR up, thanks for noticing

Reply:
Fixed the PR 🎉